core/src/main/java/org/apache/calcite/sql/SqlHopTableFunction.java

```diff
@@ -48,6 +48,9 @@ public SqlHopTableFunction() {
       if (!checkTableAndDescriptorOperands(callBinding, 1)) {
         return throwValidationSignatureErrorOrReturnFalse(callBinding, throwOnFailure);
       }
+      if (!checkTimeColumnDescriptorOperand(callBinding, 1)) {
+        return throwValidationSignatureErrorOrReturnFalse(callBinding, throwOnFailure);
```

Comment:
Why need this addition (as this PR is to change SESSION)? is there a test case verify this line of change?

Reply:
For TUMBLE/HOP table function, the descriptor must refer to a time column, I just add this check by the way. 
currently, there is no tests to verify this change, but If it's ok to do it in this PR, I can add some tests for it.
Or put it in another PR? what's your idea?

Reply:
I see. Please add tests to this PR and we can merge it together.

Reply:
Done